content/articles/create-a-headless-cms-website-using-wordpress-nextjs-and-graphql/index.md

```diff
@@ -0,0 +1,439 @@
+
+### Testing
+
+To test the above, ensure the development server is up and running, using the command;
+
+```bash
+npm run dev
+```
+
+In your browser, navigate to `http://localhost:3000`. Your post will be displayed based on the posts you added to your WordPress blog.
+
+![posts-page](/engineering-education/create-a-headless-cms-website-using-wordpress-nextjs-and-graphql/posts-page.png)
+
+### Testing with a single post
+
+To show a single post page, navigate to the the `pages` directory and create the `posts` directory. Inside the `posts` directory, create `[id].js` file. The [] signifies that the id will be dynamic. In the `[id].js` file, import the `getSinglePost` function;
+
+```js
+import {getSinglePost} from "../../lib/api";
+```
+
+This function;
+
+- Gets the posts sent from the `getServerSideProps()`.
+
+- Maps through the posts and for each post showing the title, excerpt, and the author. The title is a link to a specific posts page.
+
+Next, implement `getServerSideProps()` as follows;
+
+```js
+export async function getServerSideProps(ctx){
+    let {id} = ctx.params;
+    let post = await getSinglePost(id);
+    
+    return {
+        props:{
+            post
+        }
+    }
+}
+```
+
+From the above block of code, we get the post's id, fetching the post based on that id, and sending the post inside the props object. The next step is to implement a function for the view. To do that:
+
+Import the styles module and Head package from Next.js:
+
+```js
+import Head from "next/head";
+import styles from '../../styles/Home.module.css';
+```
+
+Create a `Post` function as below:
+
+```js
+export default function Post({post}) {
+    return (
+        <div className={styles.container}>
+
+            <Head>
+                <title>{post.title}</title>
+            </Head>
+            <main>                
+                <div dangerouslySetInnerHTML={{__html:post.excerpt}} />
+                <br />
+                <br />
+                <div dangerouslySetInnerHTML={{__html:post.content}} />
+            </main>
+
+        </div>
+    )
+}
+```
+
+Here we are simply getting the post passed and rendering it to the view from the above function by showing its title, excerpt, and content. Feel free to show as many fields as you want but make sure they are in the query.
+
+To test this, ensure the development server is running, else start it by running;
+
+```bash
+npm run dev
+```
+
+In your browser, navigate to `http://localhost:3000`.
+
+To test the implementation, click on the title of any post on the home page. You should be redirected to a single post page. Then, click on the back arrow and explore the other posts.
+
+### Conclusion
+We have managed to create posts in WordPress, query, and show them from a Next.js application. To gain more insightful knowledge about the various technologies and concepts used, check out the following resources.
```

Comment:
Here, you mention that the reader should check out the following resources, do you mean the resources on the reference section? If so, it would be able to instead write, "...check out the resources on the reference section"